pkgs/llvm_cheriot.nix

```
@@ -0,0 +1,28 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+{pkgs, ...}:
+pkgs.stdenv.mkDerivation rec {
+  name = "llvm-cheriot";
+  src = pkgs.fetchFromGitHub {
+    owner = "CHERIoT-Platform";
+    repo = "llvm-project";
+    rev = "432ef65c96c9f8be8b7802a8bfb9bcf3609b2b79";
+    sha256 = "sha256-ddLnG1olyRYxvKJ0zKEbThq/ZnxP68X32+1ws76WeV4=";
+  };
+  sourceRoot = "${src.name}/llvm";
+  nativeBuildInputs = with pkgs; [cmake ninja lld python3];
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DLLVM_ENABLE_ASSERTIONS=True"
```

```suggestion
    "-DLLVM_ENABLE_ASSERTIONS=ON"
```
I think we want to use a single convention for booleans. ON/OFF seems most common for cmake.

---

nixpkgs also has some functions you could use if you want to follow their conventions / are okay with that :

https://github.com/NixOS/nixpkgs/blob/c214b37af3abe1e99f158a8d8b59db4b709cb6e7/pkgs/tools/graphics/gmic-qt/default.nix#L95-L99

---

This is cool!